File: campuscrib/backend/microservices/crib-manager-service/src/main/java/com/campuscrib/crib_manager_service/application/ports/CribRepository.java
package com.campuscrib.crib_manager_service.application.ports;

import com.campuscrib.crib_manager_service.domain.models.Crib;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface CribRepository {
    Crib save(Crib crib);

    Optional<Crib> findById(UUID id);

    List<Crib> fetchAll(UUID landlordId);

    Boolean delete(UUID id);
}

File: campuscrib/backend/microservices/crib-search-service/domain/repository/crib_repository.go
package repository

import (
	"context"

	"crib-search-service/domain/models"

	"github.com/google/uuid"
)

type CribRepository interface {
	Save(ctx context.Context, crib *models.Crib) error
	Update(ctx context.Context, crib *models.Crib) error
	Delete(ctx context.Context, id uuid.UUID) error
	FindById(ctx context.Context, id uuid.UUID) (*models.Crib, error)
	FindAll(ctx context.Context, filters map[string]interface{}) ([]models.Crib, error)
}
